BuildScript/BuildTasks/TestConfigurationsFactory.cs

```diff
@@ -144,11 +138,51 @@ private string GetPlatform (IEnumerable<string> configurationItems)
 
     private ExecutionRuntime GetExecutionRuntime (IEnumerable<string> configurationItems)
     {
-      var kvp = _supportedExecutionRuntimes.Single (configurationItems, () => CreateMissingConfigurationItemException ("execution runtime"));
+      var enumeratedConfigurationItems = configurationItems.ToArray();
+      var intersection = _supportedExecutionRuntimes.Intersect (enumeratedConfigurationItems).ToArray();
+
+      if (intersection.Length == 0)
+      {
+        throw CreateMissingConfigurationItemException ("execution runtime");
+      }
+      else if (intersection.Length == 1)
+      {
+        return CreateExecutionRuntimeFromSingle (intersection.Single());
+      }
+      else
+      {
+        return CreateExecutionRuntimeFromMultiple (intersection, enumeratedConfigurationItems);
+      }
+    }
```

Comment:
how about:
var intersection = ...ToList(); // instead of ToArray()
bool hasLocalMachine = intersection.Remove (LocalMachine)
bool hasEnforcedLocalMachine = intersection.Remove (EnforcedlocalMachine)
if (intersection.Count > 1) throw InvalidOpEx ("multiple Docker images found")
var dockerImage = interestion.SingleOrDefault();

then do the IF/ELSEIF/ELSE

Reply:
I tried my best to reduce clutter in the method by providing some overloads and extension methods, please take a look if it is still too cluttered or maybe even _okay_